packages/eas-cli/src/commands/branch/delete.ts

```diff
@@ -0,0 +1,180 @@
+      >(
+        gql`
+          query ViewBranch($appId: String!, $name: String!) {
+            app {
+              byId(appId: $appId) {
+                id
+                updateBranchByName(name: $name) {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        `,
+        {
+          appId,
+          name,
+        }
+      )
+      .toPromise()
+  );
+
+  const {
+    app: {
+      byId: {
+        updateBranchByName: { id: branchId },
+      },
+    },
+  } = data;
+  return { branchId };
+}
+
+async function deleteBranchOnAppAsync({
+  branchId,
+}: {
+  branchId: string;
+}): Promise<DeleteUpdateBranchResult> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<
+        { updateBranch: { deleteUpdateBranch: DeleteUpdateBranchResult } },
+        {
+          branchId: string;
+        }
+      >(
+        gql`
+          mutation DeleteUpdateBranch($branchId: ID!) {
+            updateBranch {
+              deleteUpdateBranch(branchId: $branchId) {
+                id
+              }
+            }
+          }
+        `,
+        {
+          branchId,
+        }
+      )
+      .toPromise()
+  );
+  return data.updateBranch.deleteUpdateBranch;
+}
+
+export default class BranchDelete extends Command {
+  static hidden = true;
+  static description = 'Republish an update group';
+
+  static args = [
+    {
+      name: 'name',
+      required: false,
+      description: 'Name of the branch to delete',
+    },
+  ];
+  static flags = {
+    json: flags.boolean({
+      description: `return a json with the edited branch's ID and name.`,
+      default: false,
+    }),
+  };
+
+  async run() {
+    let {
+      args: { name },
+      flags: { json: jsonFlag },
+    } = this.parse(BranchDelete);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const { exp } = await getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const accountName = getProjectAccountName(exp, await ensureLoggedInAsync());
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: exp.slug,
+    });
+
+    if (!name) {
+      const validationMessage = 'branch name may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ name } = await promptAsync({
+        type: 'text',
+        name: 'name',
+        message: 'Please enter the name of the branch to delete:',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    const { branchId } = await getBranchInfoAsync({ appId: projectId, name });
+
+    if (!jsonFlag) {
+      Log.addNewLineIfNone();
+      Log.warn(
+        `You are about to permamently delete branch: "${name}" and all of the updates published on it.` +
+          `\nThis action is irreversable.`
```

Comment:
I'm not sure if we still have it in Expo CLI, but making this text might help on the emphasis?

Reply:
That's a fair point, for now will keep like this for consistency as other parts of the CLI output like this.

There is this general issue here as well: https://linear.app/expo/issue/ENG-205/standardize-eas-cli-output